onnxruntime/contrib_ops/rocm/bert/batched_gemm_softmax_gemm_permute_pipelines.cuh

```diff
@@ -416,38 +491,47 @@ auto GetCKGemmSoftmaxGemmPermuteTypeStringAndOps() {
     auto invoker = impl->MakeInvokerPointer();
     auto op = [impl = std::move(impl), invoker = std::move(invoker)](
                   const GemmSoftmaxGemmPermuteParams<T>* params) -> Status {
+      TUNABLE_OP_RETURN_UNSUPPORTED_ARGUMENT_IF(
+          !GemmSoftmaxGemmPermuteTunableOp<T>::IsSupportedQkvFormat(params->attention),
+          "qkv format is not supported, got ", params->attention->qkv_format);
       if constexpr (USE_BIAS) {
-        TUNABLE_OP_RETURN_UNSUPPORTED_ARGUMENT_IF(params->bias_buffer == nullptr);
+        TUNABLE_OP_RETURN_UNSUPPORTED_ARGUMENT_IF(
+            params->bias_buffer == nullptr, "biased version only support input with bias");
       } else {
-        TUNABLE_OP_RETURN_UNSUPPORTED_ARGUMENT_IF(params->bias_buffer != nullptr);
+        TUNABLE_OP_RETURN_UNSUPPORTED_ARGUMENT_IF(
+            params->bias_buffer != nullptr, "non-biased version only support input without bias");
       }
       if constexpr (USE_MASK) {
         TUNABLE_OP_RETURN_UNSUPPORTED_ARGUMENT_IF(
-            !GemmSoftmaxGemmPermuteTunableOp<T>::IsSupportedMaskType(params->attention));
-        TUNABLE_OP_RETURN_UNSUPPORTED_ARGUMENT_IF(params->mask_index_buffer == nullptr);
+            !GemmSoftmaxGemmPermuteTunableOp<T>::IsSupportedMaskType(params->attention),
+            "mask type is not supported, got ", params->attention->mask_type);
+        TUNABLE_OP_RETURN_UNSUPPORTED_ARGUMENT_IF(
+            params->mask_index_buffer == nullptr, "masked version only support input with mask");
       } else {
-        TUNABLE_OP_RETURN_UNSUPPORTED_ARGUMENT_IF(params->mask_index_buffer != nullptr);
+        TUNABLE_OP_RETURN_UNSUPPORTED_ARGUMENT_IF(
+            params->mask_index_buffer != nullptr, "non-masked version only support input without mask");
       }
 
       auto attn = params->attention;
-      int G0 = attn->batch_size;
-      int G1 = attn->num_heads;
-      int M = attn->sequence_length;
-      int N = attn->total_sequence_length;
-      int K = attn->head_size;
-      int O = attn->v_head_size;
+      const int& G0 = attn->batch_size;
```

Comment:
> int -> const int&

Can you please explain the rationale for this change?

Reply:
`{q,k,v}_buffer_strides` computing logic is abstracted into `GetQkvStrides`. At there I use `const&` to indicate they are merely aliases. So I also changed here to also reflect the fact.

These aliases allow us to better align the language we speak with CK developer.